autotest/gdrivers/jp2grok.py

```diff
@@ -0,0 +1,3013 @@
+    if return_error_count:
+        return (res.error_count, res.warning_count)
+    if res.error_count == 0 and res.warning_count == 0:
+        return
+    pytest.fail()
+
+###############################################################################
+# Test INSPIRE_TG support
+
+
+def test_jp2grok_26():
+
+    src_ds = gdal.GetDriverByName('MEM').Create('', 2048, 2048, 1)
+    sr = osr.SpatialReference()
+    sr.ImportFromEPSG(32631)
+    src_ds.SetProjection(sr.ExportToWkt())
+    src_ds.SetGeoTransform([450000, 1, 0, 5000000, 0, -1])
+
+    # Nominal case: tiled
+    out_ds = gdaltest.jp2grok_drv.CreateCopy('/vsimem/jp2grok_26.jp2', src_ds, options=['INSPIRE_TG=YES'])
+    overview_count = out_ds.GetRasterBand(1).GetOverviewCount()
+    # We have 2x2 1024x1024 tiles. Each of them can be reconstructed down to 128x128.
+    # So for full raster the smallest overview is 2*128
+    assert (out_ds.GetRasterBand(1).GetOverview(overview_count - 1).XSize == 2 * 128 and \
+       out_ds.GetRasterBand(1).GetOverview(overview_count - 1).YSize == 2 * 128)
+    out_ds = None
+    assert gdal.VSIStatL('/vsimem/jp2grok_26.jp2.aux.xml') is None
+    assert validate('/vsimem/jp2grok_26.jp2') != 'fail'
+    gdal.Unlink('/vsimem/jp2grok_26.jp2')
+
+    # Nominal case: untiled
+    out_ds = gdaltest.jp2grok_drv.CreateCopy('/vsimem/jp2grok_26.jp2', src_ds, options=['INSPIRE_TG=YES', 'BLOCKXSIZE=2048', 'BLOCKYSIZE=2048'])
+    overview_count = out_ds.GetRasterBand(1).GetOverviewCount()
+    assert (out_ds.GetRasterBand(1).GetOverview(overview_count - 1).XSize == 128 and \
+       out_ds.GetRasterBand(1).GetOverview(overview_count - 1).YSize == 128)
+    gdal.ErrorReset()
+    out_ds.GetRasterBand(1).Checksum()
+    assert gdal.GetLastErrorMsg() == ''
+    out_ds.GetRasterBand(1).GetOverview(0).Checksum()
+    assert gdal.GetLastErrorMsg() == ''
+    out_ds = None
+    assert gdal.VSIStatL('/vsimem/jp2grok_26.jp2.aux.xml') is None
+    assert validate('/vsimem/jp2grok_26.jp2') != 'fail'
+    gdal.Unlink('/vsimem/jp2grok_26.jp2')
+
+    # Nominal case: RGBA
+    src_ds = gdal.GetDriverByName('MEM').Create('', 128, 128, 4)
+    src_ds.SetProjection(sr.ExportToWkt())
+    src_ds.SetGeoTransform([450000, 1, 0, 5000000, 0, -1])
+    out_ds = gdaltest.jp2grok_drv.CreateCopy('/vsimem/jp2grok_26.jp2', src_ds, options=['INSPIRE_TG=YES', 'ALPHA=YES'])
+    out_ds = None
+    ds = gdal.OpenEx('/vsimem/jp2grok_26.jp2', open_options=['1BIT_ALPHA_PROMOTION=NO'])
+    assert ds.GetRasterBand(4).GetColorInterpretation() == gdal.GCI_AlphaBand
+    assert ds.GetRasterBand(4).GetMetadataItem('NBITS', 'IMAGE_STRUCTURE') == '1'
+    ds = None
+    assert gdal.VSIStatL('/vsimem/jp2grok_26.jp2.aux.xml') is None
+    assert validate('/vsimem/jp2grok_26.jp2') != 'fail'
+    gdal.Unlink('/vsimem/jp2grok_26.jp2')
+
+    # Warning case: disabling JPX
+    gdal.ErrorReset()
+    gdal.PushErrorHandler()
```

Comment:
all occurences of gdal.PushErrorHandler() / gdal.PopErrorHandler() are now replaced with the pattern:
```
with gdal.error_handler():
     foo()
```